pkgs/development/haskell-modules/configuration-ghc-8.6.x.nix

```diff
@@ -91,4 +91,7 @@ self: super: {
   # ghc versions prior to 8.8.x needs additional dependency to compile successfully.
   ghc-lib-parser-ex = addBuildDepend super.ghc-lib-parser-ex self.ghc-lib-parser;
 
+  # Only 0.6 is compatible with ghc 8.6 https://hackage.haskell.org/package/apply-refact/changelog
+  apply-refact = super.apply-refact_0_6_0_0;
```

Comment:
Thanks for trying to fix this!

However, there are a couple things you'll need to fix up before this can be merged in:

First, Haskell-related PRs should generally be based on the `haskell-updates` branch (not `master`), and they should be targetting the `haskell-updates` branch here on GitHub (not `master`).  Please change this.

Second, this `apply-refact_0_6_0_0` derivation unfortunately does not currently exist.

If you do `nix repl /some/path/to/nixpkgs`, and then type `haskellPackages.apply-refact_0_6_0_0` (or `haskell.packages.ghc865.apply-refact_0_6_0_0`, you should see a message that the derivation can't be found.

However, there is a list of extra derivations to generate here:

https://github.com/NixOS/nixpkgs/blob/d8b7b590c7b1166fad48935d757c3ff10375c89a/pkgs/development/haskell-modules/configuration-hackage2nix.yaml#L2356-L2361

If you add something like 

```yaml
- apply-refact == 0.6.0.0
- apply-refact == 0.7.0.0
```

 to this list, then these new derivations will be generated next time the Haskell packages are regenerated.

I think I might have explained a little how this works here? https://discourse.nixos.org/t/haskellpackages-stm-containers-fails-to-build/5416/4

Reply:
Thanks for the feedback. I tried to integrate it. Is the automatic generation of packages working on `haskell-updates` only or on forks of it, too?